.github/workflows/k8s-test.yml

```
@@ -0,0 +1,362 @@
+name: Kubernetes Deployment Test
+
+on:
+  pull_request:
+    paths:
+      - 'deploy/kubernetes/**'
+      - 'deploy/docker/**'
+      - '.github/workflows/k8s-test.yml'
+  schedule:
+    # Weekly on Sundays at 3 AM UTC
+    - cron: '0 3 * * 0'
+  workflow_dispatch:
+
+env:
+  DOCKER_REGISTRY: ghcr.io
+  DOCKER_IMAGE: alex-ilgayev/mcpspy
+
```

Any chance limiting the github actions permissions?
I think contents read should be enough.
```suggestion
permissions:
  contents: read
```